### ISSUE TITLE
Allow running multiple instances

### ISSUE THREAD

**Issue by rdiaz02** [sje30/ess-unigd#2]
I think this is not currently possible, because the port is hardwired in the code. Would it be possible to run `hgd(token = FALSE)` in R, grab the URL (including port) generated by the call to R in the inferior R process, and then pass that to the Perl script?

(Anyway, this is of course just a wish list. This is already great! Thanks!!)

**Comment by sje30** [comment]
Thanks Ramon.

I'm experimenting with exactly that -- and also using the ugd() interface available in R.  Watch this space!

hope you are keeping well, its been a while!

**Comment by sje30** [comment]
I'm hoping the perl script can soon be replaced by a pure elisp solution.

**Comment by rdiaz02** [comment]
Hi Stephen,
Things are going well around here. Hope you are doing well too. :-)

> I'm hoping the perl script can soon be replaced by a pure elisp solution.

Neat!! I'll be watching this closely.

**Comment by rdiaz02** [comment]
Just to keep track of a quick silly experiment. With v2, I can have different Emacs sessions use essgd. However, if I launch two different R sessions from the same Emacs session, they steal the *essgd* buffer from each other (i.e., the inferior R where I last issued `essgd-start` is the one that sends plots to the `*essgd*` buffer); doing `rename-uniquely` does not help (the renamed buffer does not seem to be used afterwards).

This might all be obvious and expected. Close as needed, of course.

**Comment by sje30** [comment]
Yes, it is as expected for now.  I hope to get it such that one Emacs session can have multiple *essgd* buffers, just like you can have multiple *R* buffers.

One step forward is however that the port and token are not hardcoded now.
